File: q1464/q1464.cpp
class Solution
{
public:
    int maxProduct(vector<int> &nums)
    {
        int firstMax = INT_MIN;
        int secondMax = INT_MIN;
        for (const int num : nums)
        {
            if (num > firstMax)
            {
                secondMax = max(firstMax, secondMax);
                firstMax = num;
            }
            else if (num > secondMax)
            {
                secondMax = num;
            }
        }
        return (firstMax - 1) * (secondMax - 1);
    }
};

File: q1464/q1464.java
class Solution {
    public int maxProduct(int[] nums) {
        int firstMax = Integer.MIN_VALUE;
        int secondMax = Integer.MIN_VALUE;
        for (int num : nums) {
            if (num > firstMax) {
                secondMax = Math.max(firstMax, secondMax);
                firstMax = num;
            } else if (num > secondMax) {
                secondMax = num;
            }
        }
        return (firstMax - 1) * (secondMax - 1);
    }
}